src/r_voxel.c

```diff
@@ -319,8 +319,25 @@ void VX_Init (void)
 
 #define VX_MINZ         (   4 * FRACUNIT)
 #define VX_MAX_DIST     (2048 * FRACUNIT)
+#define VX_MIN_DIST     ( 512 * FRACUNIT)
```

Comment:
@fabiangreffrath Have you tried nuts.wad? The `512` distance is very noticeable here. Maybe we should choose different values?

Reply:
No, not yet. Maybe the steps in maximum distance should be a bit more fine-grained? In the end, by multiplying with 2, we only have three steps: 512, 1024 and 2048.

Reply:
> Maybe the steps in maximum distance should be a bit more fine-grained? In the end, by multiplying with 2, we only have three steps: 512, 1024 and 2048.

Maybe. But usually big maps have a lot of monsters, and we will always have minimum distance and minimum resolution there.

Another problem is piles of corpses, lots of models in one place slow down the game a lot. I tried to limit the number of models, but it's too noticeable ("pop-in" like in modern games).

Reply:
We could render corpses as sprites once we reached VX_MIN_DIST.

Reply:
> We could render corpses as sprites once we reached VX_MIN_DIST.

Yes, but when voxel corpses are swapped for sprites, it looks jarring. Anyway, I'm looking for feedback, personally I like to sacrifice everything for framerate 😄

Reply:
Now you have the choice of two evils: Noticeable switches from corpse voxels to sprites for an increase in framerate, or continue rendering corpse voxels at the cost of frame rate.

Reply:
> Now you have the choice of two evils: Noticeable switches from corpse voxels to sprites for an increase in framerate, or continue rendering corpse voxels at the cost of frame rate.

Exactly, and I feel uncomfortable deciding this myself. I'm one of those people who played Quake 2 almost without textures 😆